webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {format} from 'date-fns';
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useDebouncedCallback} from 'use-debounce';
+
+import Button from 'components/common/button/button';
+import LinearScaleQuestion from 'components/surveyPost/linearScaleQuestion/linearScaleQuestion';
+import TextQuestion from 'components/surveyPost/textQuestion/textQuestion';
+
+import './style.scss';
+
+import type {
+    CustomPostTypeComponentProps,
+    MattermostWindow,
+    SurveyResponse, UserSurvey,
+} from 'types/mattermost-webapp';
+
+function SurveyPost({post, isRHS}: CustomPostTypeComponentProps) {
+    const [survey, setSurvey] = useState<UserSurvey>();
+    const [disabled, setDisabled] = useState<boolean>(false);
+    const [expired, setExpired] = useState<boolean>(false);
+    const draftResponse = useRef<SurveyResponse>();
+    const [errorMessage, setErrorMessage] = useState<string>();
+
+    const linearScaleQuestionID = useRef<string>();
+
+    const fetchUserSurvey = async (surveyID: string) => {
+        // make actual API call to fetch survey here...
+        // Replace dummy data with data fetched from API call.
+        const survey: UserSurvey = {
+            surveyId: surveyID,
+            startDate: '01/04/2024',
+            endDate: '30/04/2024',
+            questions: [
+                {
+                    id: '49fc9a85-b4d8-424e-b13f-40344b168123',
+                    mandatory: true,
+                    system: true,
+                    text: 'How likely are you to recommend Mattermost?',
+                    type: 'linear_scale',
+                },
+                {
+                    id: 'aa026055-c97c-48d7-a025-c44590078963',
+                    mandatory: true,
+                    system: true,
+                    text: 'How can we make your experience better?',
+                    type: 'text',
+                },
+                {
+                    id: '0eee4429-5083-41ef-bda7-2ee9c5ece929',
+                    mandatory: false,
+                    system: false,
+                    text: 'option question text',
+                    type: 'text',
+                },
+            ],
+            status: 'in_progress',
+
+            // response: {
+            //     responses: {
+            //         '49fc9a85-b4d8-424e-b13f-40344b168123': '8',
+            //         'aa026055-c97c-48d7-a025-c44590078963': 'Response 1',
+            //         '0eee4429-5083-41ef-bda7-2ee9c5ece929': 'Response 2',
+            //     },
+            //     dateCreated: format(new Date(), 'dd/MM/yyyy'),
+            // },
+        };
+
+        setSurvey(survey);
+
+        // the first linear scale question is the system default rating question
+        const linearScaleQuestion = survey.questions.find((question) => question.type === 'linear_scale');
```

Comment:
Just in case we add a different first question. Checking for the question type feels a bit assuring 😛

Reply:
Perhaps we could add a special flag to that one survey question to mark that it's the general NPS question? That's only needed if we plan to let users reorder the questions though so that the NPS one isn't first

Reply:
Added check for system type question along with question type.